packages/flutter/lib/src/rendering/flex.dart

```diff
@@ -1140,6 +1140,96 @@ class RenderFlex extends RenderBox with ContainerRenderObjectMixin<RenderBox, Fl
   }
 }
 
+/// A space with a fixed length in the main axis of a [RenderFlex].
+///
+/// This render must be a direct child of a [RenderFlex], or else an error
+/// will be thrown in runtime.
+///
+/// See also:
+///
+/// * [Spacer.fixed], the widget equivalent.
+final class RenderFixedLengthSpacer extends RenderBox {
+  /// Creates a render box that provides a fixed space in a render flex.
+  RenderFixedLengthSpacer({
+    required double length,
+  })  : _length = length;
+
+  /// The size of the space in the [parent]'s main axis.
+  double get length => _length;
+  double _length;
+  set length(double value) {
+    if (_length != value) {
+      _length = value;
+      markNeedsLayout();
+    }
+  }
+
+  @override
+  bool get sizedByParent => true;
+
+  @override
+  double computeMinIntrinsicWidth(double height) => 0.0;
+
+  @override
+  double computeMaxIntrinsicWidth(double height) {
+    assert(parent is RenderFlex);
+
+    return switch ((parent! as RenderFlex).direction) {
+      Axis.horizontal => length,
+      Axis.vertical => 0.0,
+    };
+  }
+
+  @override
+  double computeMinIntrinsicHeight(double width) => 0.0;
+
+  @override
+  double computeMaxIntrinsicHeight(double width) {
+    assert(parent is RenderFlex);
+
+    return switch ((parent! as RenderFlex).direction) {
+      Axis.horizontal => 0.0,
+      Axis.vertical => length,
+    };
+  }
+
+  @override
+  Size computeDryLayout(BoxConstraints constraints) {
+    assert(this.parent is RenderFlex);
```

Comment:
render objects must be agnostic regarding their parent.
this will fail in weird unpredictable ways because there's no guarantee than the parent changing state will cause the child to be dirtied
also it breaks composition because now you can't put a no-op object between the parent and child.

Reply:
What would be the proper way to solve this?

Another implementation route would be to use `ParentData`, but `ParentDataWidget` does the same kind of checking. For instance, a `Expanded` that's not a direct child of a `Flex` also throws, and it also can't be used with a no-op intermediate widget.

Reply:
Yeah ParentDataWidgets are a weird violation of the rule that widgets should be parent-agnostic. (They don't violate the rule that render objects should be parent-agnostic, because they don't map to a render object.)

I'm not sure how to implement `Spacer.fixed`, frankly, that's why it doesn't yet exist. :-)